### PR TITLE
In 136216163 format ssn

### PR DESCRIPTION
**Pull request by AmanShurafa**

### CODE REVIEW

File: app/assets/javascripts/components/common/InputField.jsx [hunk 1]
```diff
@@ -1,21 +1,33 @@
-import React from 'react'
+import MaskedInput from 'react-maskedinput'
 import PropTypes from 'prop-types'
+import React from 'react'
+import _ from 'lodash'
+
+const InputField = ({gridClassName, labelClassName, id, label, onChange, value, placeholder, type, maxLength, mask}) => {
```
Comment (saraOSI) [inline app/assets/javascripts/components/common/InputField.jsx]: The fact that `InputField` now renders one of two different objects is a bit unclear from the way we call it on other components. Is there a way we can make it more clear to the developers what is going on without having to reference the component?

File: app/assets/javascripts/utils/ssnFormatter.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import _ from 'lodash'
+
+const ssnFormatter = (ssn) => {
+  if (_.isEmpty(ssn)) {
+    return ''
+  }
+  // eslint-disable-next-line no-magic-numbers
```
Comment (saraOSI) [inline app/assets/javascripts/utils/ssnFormatter.js]: Why disable no magic numbers here? Setting up a const is fairly easy, and will suppress the warning.
```
cont completeSSNLength = 9
if (ssn.length === completeSSNLength && !ssn.includes('-')) {
...
```

File: spec/features/screening/participant/editing_participant_spec.rb [hunk 3]
```diff
@@ -152,6 +152,33 @@
...
+          expect(page).to have_field('Social security number', with: '')
+          fill_in 'Social security number', with: 1
+          expect(page).to have_field('Social security number', with: '1__-__-____')
+          fill_in 'Social security number', with: 123_456_789
+          expect(page).to have_field('Social security number', with: '123-45-6789')
+        end
+      end
+    end
+
+    scenario 'an invalid character is inserted' do
```
Comment (saraOSI) [inline spec/features/screening/participant/editing_participant_spec.rb]: Any other invalid characters we should test? Symbols, etc? If `MaskedInput` handles the formats, then I trust the plugin to have been tested, and we can probably stick with this one negative test.

File: spec/javascripts/components/common/InputFieldSpec.jsx [hunk 4]
```diff
@@ -59,4 +59,67 @@ describe('InputField', () => {
...
+      expect(inputElement.props().mask).toEqual('111-111-1111')
+    })
+  })
+
+  describe('when no validation function passed in', () => {
+    beforeEach(() => {
+      component = shallow(<InputField {...props}/>)
+    })
+
+    it('does not display error class', () => {})
```
Comment (saraOSI) [inline spec/javascripts/components/common/InputFieldSpec.jsx]: There is no test here, nor below.

File: spec/javascripts/utils/ssnFormatterSpec.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import ssnFormatter from 'utils/ssnFormatter'
+
+describe('ssnFormatter', () => {
+  it('process undefined', () => {
```
Comment (saraOSI) [inline spec/javascripts/utils/ssnFormatterSpec.js]: 'processes' might be more grammatically correct here (and elsewhere in this file) than 'process'.

File: spec/javascripts/components/common/InputFieldSpec.jsx (hunk 4 above)
Comment (AmanShurafa) [inline spec/javascripts/components/common/InputFieldSpec.jsx]: We've fixed this.

File: app/assets/javascripts/utils/ssnFormatter.js (hunk 2 above)
Comment (AmanShurafa) [inline app/assets/javascripts/utils/ssnFormatter.js]: Done

File: spec/features/screening/participant/editing_participant_spec.rb (hunk 3 above)
Comment (AmanShurafa) [inline spec/features/screening/participant/editing_participant_spec.rb]: Done

File: spec/javascripts/utils/ssnFormatterSpec.js (hunk 5 above)
Comment (AmanShurafa) [inline spec/javascripts/utils/ssnFormatterSpec.js]: Done

File: app/assets/javascripts/components/common/InputField.jsx (hunk 1 above)
Comment (AmanShurafa) [inline app/assets/javascripts/components/common/InputField.jsx]: We agree with your comment, but feel that the functionality is almost identical and keeping it this way is more conventional. We tried to always pass Masked input but it requires a mask.

**Pull request merged**